Jenkinsfile

```
@@ -88,99 +91,50 @@ spec:
   }
 
   stages {
-
-    stage('Checkout www repo (master)') {
-      when {
-          anyOf {
-            branch 'main';
-            branch 'master';
-            branch 'publication'
-          }
-        beforeAgent true
-      }
-      steps {
-        milestone 1
-        container('jnlp') {
-          dir('www') {
-            sshagent(['git.eclipse.org-bot-ssh']) {
-                sh '''
-                    GIT_SSH_COMMAND="ssh -o StrictHostKeyChecking=no" git clone ssh://genie.${PROJECT_NAME}@git.eclipse.org:29418/www.eclipse.org/${PROJECT_NAME}.git .
-                    git checkout master
-                '''
-            }
-          }
-        }
-        milestone 2
-      }
-    }
-
     stage('Build che-docs website') {
       when {
-          anyOf {
-            branch 'main';
-            branch 'master';
-            branch 'publication'
-          }
+        anyOf {
+          branch 'main'
+          branch 'master'
+          branch 'publication'
+        }
         beforeAgent true
       }
       steps {
         milestone 21
         container('che-docs') {
           dir('che-docs') {
-                sh '''
-                git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
-                git fetch
-                ./tools/publication.sh
-                '''
-            }
+            sh './tools/build-for-publication.sh'
+          }
         }
+        milestone 22
```

what does this do?

---

It's a fail fast feature in Jenkins: if a later build is running on the same branch, and reached a higher milestone, then the current build stops directly at milestone check. 

See: https://www.jenkins.io/doc/pipeline/steps/pipeline-milestone-step/#pipeline-milestone-step